include/osquery/config.h

```diff
@@ -224,7 +239,7 @@ class ConfigPlugin : public Plugin {
    * indicates that config retrieval was successful, then the config data
    * should be returned in pair.second.
    */
-  virtual std::pair<osquery::Status, std::string> genConfig() = 0;
+  virtual Status genConfig(std::map<std::string, std::string>&) = 0;
```

Comment:
nit, it's nice to put the names of the arguments in the declarations. Even though they don't do anything for the function signatures.